src/main/scala/sbtsonar/SonarPlugin.scala

```diff
@@ -89,6 +116,22 @@ object SonarPlugin extends AutoPlugin {
       .toSeq
       .flatten
 
+  private[sbtsonar] def readPropertiesFile(
+    baseDir: File,
+    fileName: String,
+    version: String
+  ): Map[String, String] = {
+    val properties = new java.util.Properties()
+    val fileStream = new java.io.FileInputStream(new java.io.File(fileName))
+    try {
+      properties.load(fileStream)
+    } finally {
+      fileStream.close()
+    }
+
+    return properties.asScala.toMap
+  }
+
```

Comment:
Calling `close` can throw, so you should also try to handle this and ideally catch and log `NonFatal` exceptions for both calls to load and close.

It would be good if you could use more idiomatic `Try` instead or if you're feeling adventurous, you could try `Using` from sbt for safer resource management, but I think it was moved to `sbt.io` in 1.x, so you would need to provide a different implementation for 0.13 and 1.x.
https://www.scala-sbt.org/0.13.15/api/index.html#sbt.Using
https://www.scala-sbt.org/1.1.6/api/sbt/io/Using.html

A unit test would also be nice to cover this function.
The `return` at the end is redundant.